tensorflow/compiler/xla/service/allocation_tracker.h

```diff
@@ -63,10 +63,10 @@ class Allocation {
     CHECK_GE(ref_count_, 0);
     return ref_count_;
   }
-  void increment_ref_count() {
+  void increment_ref_count(int inc) {
     CHECK_GT(ref_count_, 0);
-    CHECK_LT(ref_count_, INT_MAX);
-    ++ref_count_;
+    CHECK_GT(ref_count_+inc, 0);
```

Comment:
Assuming ref_count and inc are both non-negative, this CHECK_GT will only fail in the event of signed overflow which is undefined behavior so this CHECK can't be relied upon. How about replacing it with:

CHECK_GT(inc, 0);
CHECK_LT(ref_count_, INT_MAX - inc + 1);